contracts/BancorAdaptor.sol

```diff
@@ -39,14 +39,18 @@ contract BancorAdaptor {
 /*
 return price per second of product in DATA, or reverts() if none found
 */
-	function getProductPrice(Marketplace mkt, bytes32 productId) internal returns (uint) {
+	function _getPricePerSecond(Marketplace mkt, bytes32 productId) internal returns (uint) {
 		(string memory name, address owner, address beneficiary, uint pricePerSecond, Marketplace.Currency priceCurrency, uint minimumSubscriptionSeconds, Marketplace.ProductState state) = mkt.getProduct(productId);
 		require(owner != 0x0, "not found");
-		require(priceCurrency == Marketplace.Currency.DATA, "price not in DATA");
-		return pricePerSecond;
+		if(priceCurrency == Marketplace.Currency.DATA){
+			return pricePerSecond;
+		}
+		else{
+			return pricePerSecond.mul(mkt.dataPerUsd()).div(10**18);
```

Comment:
Noticed there's a utility function `getPriceInData` on the marketplace  which you could use instead of duplicating the math and the if-else logic?

`getPriceInData(1, pricePerSecond, priceCurrency)`

(first argument is subscription length in seconds, here we want price per second so passing 1)

Reply:
ok I'll update

Reply:
done